core/runtime-dev/src/main/java/io/quarkiverse/flow/devui/WorkflowRPCService.java

```diff
@@ -25,13 +29,68 @@ public List<WorkflowInfo> getWorkflows() {
 
     @JsonRpcDescription("Generate mermaid definition from Workflow Definition")
     public MermaidDefinition generateMermaid(String name) {
-        WorkflowDefinition workflowDefinition = Arc.container().listAll(WorkflowDefinition.class)
+        WorkflowDefinition workflowDefinition = findWorkflowDefinitionByName(name);
+        return new MermaidDefinition(new Mermaid().from(workflowDefinition.workflow()));
+    }
+
+    @JsonRpcDescription("Get Workflow input specification")
+    public WorkflowInput getInputSpecification(String workflowName) {
+        WorkflowDefinition workflowDefinition = findWorkflowDefinitionByName(workflowName);
+
+        if (notContainsSchema(workflowDefinition)) {
+            return new WorkflowInput(false, null);
+        }
+
+        return new WorkflowInput(true, workflowDefinition.workflow().getInput().getSchema());
+    }
+
+    @Blocking
+    @JsonRpcDescription("Execute Workflow given an input")
+    public WorkflowOutput executeWorkflow(
+            @JsonRpcDescription("Workflow name") String workflowName,
+            @JsonRpcDescription("Workflow input as JSON format") Map<String, Object> inputJson,
+            @JsonRpcDescription("Workflow input as text format") String inputText) {
+
+        WorkflowDefinition workflowDefinition = findWorkflowDefinitionByName(workflowName);
+
+        Object input;
+        if (inputJson != null) {
+            input = inputJson;
+        } else if (inputText != null) {
+            input = inputText;
+        } else {
+            input = "";
+        }
+
+        WorkflowModel wm = workflowDefinition.instance(input)
+                .start()
+                .join();
+
+        if (wm.asJavaObject() instanceof String) {
+            return new WorkflowOutput("text/plain", wm.asText());
+        }
```

Comment:
I think it worth to try removing this one after performing the change below. 
If not working and it is really needed, then, you can do this
```
Object obj = wm.asJavaObject() ;
return new WorkfklowOuput(obj instanceof String ? "text/plain": "application/json", obj);
```
I think WorkflowOutput admits object and obj is already a string or a map, so you do not need to call wm again